versioning/versioning.md

```diff
@@ -0,0 +1,113 @@
+
+![Alt](images/MobileServiceIntegrationMatrix.png)
+
+**Mobile Service Integration Version Compatiability Matrix**
+
+Once a developer knows what services work together, then they need to know what versions of those services work together, they need a version compatiability matrix. See Example below;
+
+![Alt](images/MobileServiceIntegVerCompatibilityMatrix.png)
+
+### Process
+With regard to the versioning process what are the necessary changes/points to note;
+
+* If a sub-component of a service changes
+  * That sub-component is stepped and the overall Service version is stepped
+  * The other sub-components of the Service are not stepped
+
+  ![Alt](images/MobileServiceMatrix-SynchEx.png)
+
+* Release Notes should be produced at a Mobile Service level
+  * You can see what sub-components changed.
+* You can patch a sub-component, e.g. SDK-Android-Sync 0.4.0 -> 0.4.1
+  * The patched component will need to be tested with other components within the service which it interacts with (i.e. its touch points)
+  * Based on tests passing; the overall service version will not change, but the patch is compatible. SDK-Android-Sync 0.4.1 is compatible with Sync 0.4 and all its constituent components.
+* If the Mobile Service which is being offered by AeroGear makes use of another standalone backend Service, the version of that Service must be represented
+  * Keycloak Service is not under Mobile/Aerogear’s control but its a primary component of AeroGear's Auth Service. We package it along with an APB and SDKs. We need to highlight what version of Keycloak is being used inside our different versions of our Auth Service. See Example below;
+
+  ![Alt](images/MobileServiceAuthExample.png)
+
+
+### Releasing
+All components can be released separately, but it may be worth considering the release of a Mobile Serice after every Sprint or second Sprint. In doing that you are producing a compatiable set of components which represents the service.
+
+
+## Additional Considerations
+- the relationship between the Core SDK and the Service SDKs from a version perspective! This relationship will need to be handled but the developer more than likley will not see the Core SDK or not have to worry about it.
+- How much defensive functionality is developed around allowing or disallowing Mobile Service Integrations to be requested if we know that there will be compatiability issues.
+  - this could be addressed in the IDE Plugin or the Mobile CLI. It possibly doesn't need to be addressed now but can be considered in the future.
+- the following will most likely also need versions
+  - Mobile Core (installer?)
+     - the installer will point to the repo where the playbooks are installed, currently aerogear/playbooks
+     - in downstream, it may point at a different organisation
+  - MiniShift Plugin
+  - Mobile App Representation (their APBs) - do they have a dependency on a Mobile Service, possibly CI/CD service? Worth considering
```

Comment:
@maleck13 Do you think there will be a need to version 'Mobile App REpresentations'? In that they are created from a Custom Resource Definition which is created by the Mobile-Core; do the Mobile App Representations just fall in under the version of Mobile-Core and we don't have to worry about them as individual components which need their own version?

Reply:
the mobile clients are a CRD so are versioned naturally as per the general kubernetes versioning system
I don't see there being an impact on mobile services, I think it will only be the mobile cli, IDE and UI that use these directly